tests/model/Megatron_GPT2/ds_config_func_bs4_zero1.json

```diff
@@ -3,14 +3,9 @@
   "gradient_accumulation_steps": 1,
   "steps_per_print": 1,
   "zero_optimization": {
-    "stage":1
-  },
-  "optimizer": {
-    "type": "Adam",
-    "params": {
-      "lr": 0.00015
-    }
+    "stage": 1
   },
+  "zero_allow_untested_optimizer": true,
```

Comment:
I assume that since we're given a client provided optimizer we don't know that it's an Adam-based optimizer to know that it's safe for zero. Could we change the logic in the engine to run `isinstance(client_optimizer, apex.optimizers.fused_adam.FusedAdam)`? Then the Megatron json wouldn't need this untested optimizer flag. Also I think ZeRO would work with `torch.optim.Adam` as well or maybe not, I don't recall.

Also /cc @ShadenSmith, he mentioned to me having issues with our optimizer logic recently with PP.